clicommand/artifact_upload.go

```diff
@@ -103,6 +112,21 @@ var ArtifactUploadCommand = cli.Command{
 			Usage:  "A specific Content-Type to set for the artifacts (otherwise detected)",
 			EnvVar: "BUILDKITE_ARTIFACT_CONTENT_TYPE",
 		},
+		cli.BoolFlag{
+			Name:   "glob-resolve-follow-symlinks",
+			Usage:  "Follow symbolic links to directories while resolving globs. Note: this will not prevent symlinks to files from being uploaded. Use --upload-skip-symlinks to do that",
+			EnvVar: "BUILDKITE_AGENT_ARTIFACT_GLOB_RESOLVE_FOLLOW_SYMLINKS",
+		},
+		cli.BoolFlag{
+			Name:   "upload-skip-symlinks",
+			Usage:  "After the glob has been resolved to a list of files to upload, skip uploading those that are symlinks to files",
+			EnvVar: "BUILDKITE_ARTIFACT_UPLOAD_FOLLOW_SYMLINKS",
```

Comment:
the name and the envar are disagreeing here; one of them is skip, the other is follow. i'd default to them both being follow, given the prior art with glob-resolve, but i'm not super fussed

Reply:
upload-follow-symlinks was an early draft. I've replaced (and negated) it to upload-skip-symlinks